New pages: row-rule, *-style, and *-width (#44725)
mdn/content: files/en-us/web/css/reference/properties/row-rule-style/index.md

```

```css interactive-example-choice
row-rule-style: repeat(2, dashed, dotted), solid;
```

```css interactive-example-choice
row-rule-style: solid, repeat(auto, dashed, dotted), solid;
```

```css interactive-example-choice
row-rule-style: hidden;
```

```html interactive-example
<section id="default-example">
  <ul id="example-element">
    <li>One fish</li>
    <li>Two fish</li>
    <li>Red fish</li>
    <li>Blue fish</li>
  </ul>
</section>
```

```css interactive-example
#example-element {
  display: flex;
  flex-flow: column;
  row-rule-width: thick;
  row-rule-color: magenta;
  gap: 7px;
  text-align: left;
}
```

## Syntax

```css
/* One value */
row-rule-style: none;
row-rule-style: hidden;
row-rule-style: dotted;

/* Multiple values */
row-rule-style: groove, dashed, solid;
row-rule-style: double, repeat(5, ridge), double;
row-rule-style: solid, repeat(auto, inset, outset), solid;

/* Global values */
row-rule-style: inherit;
row-rule-style: initial;
row-rule-style: revert;
row-rule-style: revert-layer;
row-rule-style: unset;
```

### Values

The `row-rule-style` property accepts a comma-separated list of values, including:

- `<line-style>`

>   - : A {{cssxref("&lt;line-style&gt;")}}, either `none`, `hidden`, `dotted`, `dashed`, `solid`, `double`, `groove`, `ridge`, `inset`, or `outset`. The default value is`none`.

Reviewer: ```suggestion
  - : A {{cssxref("&lt;line-style&gt;")}}: one of `none`, `hidden`, `dotted`, `dashed`, `solid`, `double`, `groove`, `ridge`, `inset`, or `outset`. The default value is`none`.
```

- `<repeat-line-style>`

>   - : A {{cssxref("repeat()")}} function, with the first parameter being an {{cssxref("&lt;integer&gt;")}} of `1` or more, representing the number of times the second parameter, a comma-separated list of {{cssxref("&lt;line-style&gt;")}} values, should be repeated.

Reviewer: ```suggestion
  - : A {{cssxref("repeat()")}} function, with the first argument being an {{cssxref("&lt;integer&gt;")}} of `1` or more, and subsequent arguments being {{cssxref("&lt;line-style&gt;")}} values. The integer specifies how many times the `<line-style>` values should be repeated.
```

- `<auto-repeat-line-style>`
  - : A {{cssxref("repeat()")}} function, with `auto` as the first argument and one or more `<line-style>` values as subsequent arguments. The provided `<line-style>` values are repeated as many times as needed to fill in values for any row-rules that are not explicitly specified by other components of the property value.

## Description

The `row-rule-style` property defines the line style of any row rule lines drawn in the gaps between rows in [multi-column](/en-US/docs/Web/CSS/Guides/Multicol_layout), [flex](/en-US/docs/Web/CSS/Guides/Flexible_box_layout), and [grid](/en-US/docs/Web/CSS/Guides/Grid_layout) containers with more than one row.

> The value is a comma-separated list of components, which can include `<line-style>`, `<repeated-line-style>`, and `<auto-repeat-line-style>` types.

Reviewer: ```suggestion
The value is a comma-separated list of components, which can include `<line-style>`, `<repeat-line-style>`, and `<auto-repeat-line-style>` types.
```

The `row-rule-style`, along with the {{cssxref("row-rule-color")}} and {{cssxref("row-rule-width")}} properties, can be set using the {{cssxref("row-rule")}} shorthand. The `row-rule-style`, along with the {{cssxref("column-rule-style")}} property, can also be set using the {{cssxref("rule-style")}} shorthand.

If the property value has only one `<line-style>`, all the row rules will be that style. If we declare the following, all row rules will be `dashed`:

```css

> row-rule-style: `dashed`;

Reviewer: ```suggestion
row-rule-style: dashed;
```

```

When more than one `<line-style>` is declared, they will be applied to row-rules in the order specified. If there are more row-rules than `<line-style>` values, the list of line styles is repeated until every row rule has a style. If we declare the following, for example, every odd rule will be `dashed`, and every even rule will be `dotted`.

```css
row-rule-style: dashed, dotted;
```

### Repeated line styles

> The `repeat()` function, with an integer of `1` or greater as the first argument, can be used to repeat a valid list of CSS {{cssxref("&lt;line-style&gt;")}} values passed as the second argument the specified number of times. This allows the same style to be repeated a set number of times without repeating the same value. You can use any of the `<line-style>` keywords or custom properties that resolve to a valid `<line-style>`. Using `repeat()` can make values easier to write, enabling recurring patterns to be written using a single function, regardless of the number of rows. The following declarations are equivalent:

Reviewer: ```suggestion
The `repeat()` function, with an integer of `1` or greater as the first argument, can be used to repeat a valid list of CSS {{cssxref("&lt;line-style&gt;")}} values passed as subsequent arguments the specified number of times. This allows the same style to be repeated a set number of times without repeating the same value. You can include `<line-style>` keyword values or custom properties that resolve to a valid `<line-style>`. Using `repeat()` can make values easier to write, enabling recurring patterns to be written using a single function, regardless of the number of rows. The following declarations are equivalent:
```

```css
row-rule-style: solid, outset, inset, outset, inset;
row-rule-style: solid, repeat(2, outset, inset);
```

This creates a list of five styles. If the number of styles in the `row-rule-style` value's style list exceeds the number of gaps between rows, the excess style values are ignored. If the container has three rows, the rule in the first gutter will be `solid` and the second `outset`.

> If there are more gutters than styles, as there are no `repeat()` functions that include the keyword `auto`, the list of styles is repeated. If the container has 6, 11, 16 or 21 rows, this sequence of styles will be repeated one, two, three, or four times, respectively, with the last rule being `inset`.

Reviewer: ```suggestion
If there are more gutters than styles, the list of styles is repeated. If the container has 6, 11, 16, or 21 rows, this sequence of styles will be repeated one, two, three, or four times, respectively, with the last rule being `inset`.
```

### Auto-repeating line styles

> The `repeat()` function also accepts `auto` as the first argument instead of a positive integer. With `auto` as the first argument, the comma-separated list of `<line-style>` values passed as the second parameter will be repeated as many times as needed to fill in values for any row-rules that are not explicitly specified by other components of the property value, if any.

Reviewer: ```suggestion
The `repeat()` function also accepts `auto` as the first argument instead of a positive integer. With `auto` as the first argument, the `<line-style>` values passed as subsequent parameters will be repeated as many times as needed to fill in values for any row-rules that are not explicitly specified by other components of the property value.
```

```css
row-rule-style: solid, repeat(auto, dotted), solid;
```

> In this case, the first and last row rules will be `sold`, and all others will be `dotted`. It doesn't matter if the container has 3, 6, 11, 16, or 21 rows, the first and last rows will always be `thin`. All the other row rules will be `dotted`, which means if there are only 2 or 3 rows, there will be no dotted row rules.

Reviewer: ```suggestion
In this case, it doesn't matter if the container has 3, 6, 11, 16, or 21 rows; the first and last row rules will always be `solid`, and all the other row rules will be `dotted`. If there are only 2 or 3 rows, there will be no dotted row rules.
```

  <li>Aaron Judge + Giancarlo Stanton</li>
  <li>LeBron James + Dwyane Wade</li>
  <li>Xavi Hernandez + Andres Iniesta</li>
  <li>Kerri Walsh + Misty May Treanor</li>
</ul>
```

#### CSS

We define the list to be a flex container, creating rows by setting the {{cssxref("flex-direction")}} to `column` using the {{cssxref("flex-flow")}} shorthand. We include a {{cssxref("gap")}} of `5px` to provide enough room between the rows to fit our `3px` dashed red rule:

```css live-sample___basic live-sample___repeat live-sample___func live-sample___auto
ul {
  display: flex;
  flex-flow: column;
  gap: 5px;
  row-rule-width: 3px;
  row-rule-color: red;

  row-rule-style: dashed;
}
```

#### Result

{{EmbedLiveSample("Basic", "", "180")}}

### Repeating values

This example demonstrates how, when there are fewer values in the list of styles than row rules, the values are repeated.

Using the same HTML and CSS as in the previous example, we include three comma-separated styles as the `row-rule-style` value:

```css live-sample___repeat
ul {
  row-rule-style: solid, dotted, dashed;
}
```

{{EmbedLiveSample("Repeat", "", "180")}}

### Using the `repeat()` function

> This example demonstrates using the `repeat()` function within the `row-rule-style` property value. We use the same HTML and CSS as in the previous examples. We include a `repeat()` function, setting the list of two `<line-style>` values to be repeat 3 times.

Reviewer: ```suggestion
This example demonstrates using the `repeat()` function within the `row-rule-style` property value. We use the same HTML and CSS as in the previous examples. We include a `repeat()` function, setting the list of two `<line-style>` values to be repeated 3 times.
```

```css live-sample___func live-sample___auto
ul {
  --base: 0.5vw;
  --secondary: 1vw;

>   row-rule-style: solid, repeat(3, inset, outset), solid;

Reviewer: It's very difficult to see the difference between `solid`, `inset`, and `outset`, at least in this example. Maybe use `dotted` and `dashed` instead of `inset` and `outset`?

>   row-rule-style: solid, repeat(3, inset, outset), solid;

Author: ```suggestion
  row-rule-style: double, repeat(3, inset, dashed), double;
```
